jb/itch5/compute_book.hpp

```diff
@@ -301,26 +311,31 @@ class compute_book {
                       << ", msg=" << msg;
       return;
     }
-    auto& book = books_[position->second.stock];
-    // ... update the order list and book, but do not make a callback ...
-    auto update = do_reduce(
-        position, book, recvts, msgcnt, msgoffset, msg.header,
-        msg.original_order_reference_number, 0);
-    // ... now we need to insert the new order ...
-    orders_.emplace(
-        msg.new_order_reference_number,
-        order_data{update.stock, update.buy_sell_indicator, msg.price,
-                   msg.shares});
-    (void)book.handle_add_order(
-        update.buy_sell_indicator, msg.price, msg.shares);
-    // ... adjust the update data structure ...
-    update.cxlreplx = true;
-    update.oldpx = update.px;
-    update.oldqty = -update.qty;
-    update.px = msg.price;
-    update.qty = msg.shares;
-    // ... and invoke the callback ...
-    callback_(msg.header, book, update);
+    // ... find the right book for this order
+    // ... the book has to exists, since the original add_order created
+    // one if needed
+    auto itbook = books_.find(position->second.stock);
+    if (itbook != books_.end()) {
```

Comment:
Ack. I was lilke that, I changed because I could not find a way to write a test to cover the "return" part. If coverage report is not a problem i can use the standard way we have: throw feed_error exception. 
But, again, can you envision a test case I can write?